### DISCUSSION TITLE
Import Users to MongoDB and then link it with OneLogin (when user logins) - NextAuth v4

### DISCUSSION THREAD

**Discussion by heylols** [nextauthjs/next-auth#3263] [Help]
### Question 💬

I have `OneLogin` setup and working fine and uses MongoDB as database. But the problem is, I have some existing Users(from my previous PHP application) which I will add(import) to this database. But I want to automatically link when the actual User(that I imported) logins via OneLogin to my app for the first time.

Am I stuck with this linking part. Any thoughts? `email` would be the field that would act as the bridge. I mean for the imported Users with the actual User logging in.

### How to reproduce ☕️

..

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by anguskeatinge** [comment]
I am facing something similar at the moment. 

NextAuth doesn't support automatically linking, instead the user needs to be logged in before selecting the oauth `signIn` and then it links them for you.

I would probably do something like so:

1. importing those users into your database
2. allowing them to sign in with the `EmailProvider` (i.e. magic link)
3. once they are signed in, prompt them to click the `OneLogin` button to link that to the database user. `signIn('onelogin')` on the client.

After they click the button:

1. They will go through the onelogin flow
2. If the correct email is associated with that OneLogin account, nextauth will set the userId field on the account object in the database
3. Bob's your uncle

**Reply by balazsorban44** [reply]
Just to clarify, not supporting account linking when the user is not logged in is a deliberate choice to increase security, and is often overlooked by other solutions.

https://github.com/nextauthjs/next-auth/blob/86baefdd9df9246669d4f788aaab4774605c256a/src/server/lib/callback-handler.js#L184-L200
